manifests/vhost/docroot.pp

```diff
@@ -0,0 +1,22 @@
+
+define apache::vhost::docroot (
+  Variant[Boolean,String] $docroot,
+  $virtual_docroot                                                                  = false,
```

Comment:
Maybe it's out of scope for this PR, but I think this could be a ```Boolean``` for additional type safety (which this patch series allows for)